src/components/PostComment.jsx

```diff
@@ -0,0 +1,42 @@
+import {useState, useContext} from "react"
+import "./postComment.css"
+import {UserContext} from "../contexts/User"
+import { Login } from "../contexts/Login"
+import { postComment } from "../api"
+function PostComment({article_id, setRenderComments, renderComments, setFailedToPost}){
+    const [comment, setComment] = useState("")
+    const {user} = useContext(UserContext)
+    const {setLogin} = useContext(Login)
+    const submitHandle = (e) => {
+        e.preventDefault()
+        setFailedToPost(false)
+        if(!user.isLoggedIn) {
+            setLogin(true)
+        } else {
+            postComment(user.username, comment, article_id).then((res) => {
+                renderComments ? setRenderComments(false): setRenderComments(true)
+                setComment("")
+            }).catch(() => {
+                console.log("failed to comment")
+                setFailedToPost(true)
+            })
+        }
+    }
+
+    return (
+        <div>
+            <form className="comment__form" onSubmit={submitHandle}>
+                <input
+                    type="text"
+                    className="comment__field" 
+                    placeholder="leave a comment"
+                    value={comment}
+                    onChange={(e) => {setComment(e.target.value)}}
+                    required></input>
+                <button type="submit" >post</button>
```

Comment:
Following on from a previous comment:
"This function won't prevent a user from clicking submit multiple times on a slow connection where the first request hasn't been completed yet. Stop the user from being able to submit multiple times"

Your button needs to have a disabled={TRUE/FALSE} property so that once it is pressed it is disabled (preventing people from sending multiple requests), and then re-enabled after that query has sucessfully gone through to the api